src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -252,6 +254,54 @@ private function validateAndDenormalize($currentClass, $attribute, $data, $forma
                 $data = [$data];
             }
 
+            // In XML and CSV all basic datatypes are represented as strings, it is e.g. not possible to determine,
+            // if a value is meant to be a string, float, int or a boolean value from the serialized representation.
+            // That's why we have to transform the values, if one of these non-string basic datatypes is epxected.
+            //
+            // This is special to xml and csv format
+            if (
+                (XmlEncoder::FORMAT === $format || CsvEncoder::FORMAT === $format) &&
+                \in_array($type->getBuiltinType(), [Type::BUILTIN_TYPE_BOOL, Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_FLOAT]) &&
+                \is_string($data)
+            ) {
+                switch ($type->getBuiltinType()) {
+                    case Type::BUILTIN_TYPE_BOOL:
+                        // according to https://www.w3.org/TR/xmlschema-2/#boolean, valid representations are "false", "true", "0" and "1"
+                        if ('false' === $data || '0' === $data) {
+                            $data = false;
+                        } elseif ('true' === $data || '1' === $data) {
+                            $data = true;
+                        }
```

Comment:
what should happen in the remaining implicit else case?

Reply:
throw validation exception. An exception will probably be thrown anyway so if the property is boolean and the value in the data is "cat" might as well throw a helpful exception here.

Reply:
Type enforcement will do its thing later on (if it's enabled). Otherwise, if you're using strict types, you'd get an error; if not, nothing happens.

Reply:
If type enforcement is enabled, a `NotNormalizableValueException` will be thrown below. Otherwise: if using strict types, you might get an error; if not, nothing special (just as before).